docs/core/testing/unit-testing-mstest-writing-tests-attributes.md

```diff
@@ -353,6 +353,17 @@ The [DoNotParallelize](<xref:Microsoft.VisualStudio.TestTools.UnitTesting.DoNotP
 > [!NOTE]
 > By default, MSTest runs tests in sequential order so you only need to use this attribute if you have applied the `[Parallelize]` attribute at the assembly level.
 
+### `RetryBaseAttribute`
+
+The `RetryBaseAttribute` was introduced in MSTest 3.8. It's an abstract attribute that encapsulates a retry logic to be used when a test fails or timeouts. Implementors of this attribute must implement `ExecuteAsync` method which takes `RetryContext` as a parameter. Whenever a test times out or fails and an implementation of `RetryBaseAttribute` is present on the test method, MSTest will call `ExecuteAsync` of the attribute.
+
+> [!NOTE]
+> `RetryBaseAttribute.ExecuteAsync` is currently marked as experimental API.
+>
+> [!NOTE]
+> MSTest provides a built-in implementation of `RetryBaseAttribute`, named `RetryAttribute`.
+> The built-in `RetryAttribute` allows to control the maximum number of retry attempts, time delay between retries, and a delay backoff type which is either constant or exponential. You can check out the implementation of `RetryAttribute` on [GitHub](https://github.com/microsoft/testfx/blob/27e0f3da163f3cd651e34395c8759772012b55f9/src/TestFramework/TestFramework/Attributes/TestMethod/RetryAttribute.cs) to have better understanding of how to implement your own custom `RetryBaseAttribute`.
```

Comment:
I would probably do the opposite and describe here `RetryAttribute` that is what is useful to end users. We can mention `RetryBaseAttribute` in a note saying that it allows for user-provided attributes or we could just not mention it here and add a new documentation page about MSTest extensibility.

Reply:
@Evangelink Does it look better now?

Reply:
Maybe drop the first note

Reply:
@Evangelink Which note? About experimental? That one is already dropped

Reply:
Just me not remembering that GH isn't updating the code linked to comments :)